ai-flaw-report-streamlit.py

```diff
@@ -446,34 +461,564 @@ def display_hazard_fields():
+        }, f, indent=4)
+    
+    return file_path
+
+def get_report_from_database(report_id):
+    """Retrieve a report from the database by its ID"""
+    conn = create_db_connection()
+    if conn is None:
+        return None
+    
+    try:
+        with conn.cursor() as cur:
+            cur.execute("""
+                SELECT report_data, machine_readable_data 
+                FROM flaw_reports 
+                WHERE report_id = %s;
+            """, (report_id,))
+            
+            result = cur.fetchone()
+            if not result:
+                return None
+                
+            report_data, machine_readable_data = result
+            return {
+                "form_data": report_data,
+                "machine_readable": machine_readable_data
+            }
+    except psycopg2.Error as e:
+        st.error(f"Database error when retrieving report: {e}")
+        return None
+    finally:
+        conn.close()
+
+def get_report_recipients_from_database(report_id):
+    """Retrieve recipients for a report from the database"""
+    conn = create_db_connection()
+    if conn is None:
+        return []
+    
+    try:
+        with conn.cursor() as cur:
+            cur.execute("""
+                SELECT recipient_name, recipient_type, recipient_contact, status
+                FROM report_recipients 
+                WHERE report_id = %s;
+            """, (report_id,))
+            
+            recipients = []
+            for row in cur.fetchall():
+                recipients.append({
+                    "name": row[0],
+                    "type": row[1],
+                    "contact": row[2],
+                    "status": row[3]
+                })
+                
+            return recipients
+    except psycopg2.Error as e:
+        st.error(f"Database error when retrieving recipients: {e}")
+        return []
+    finally:
+        conn.close()
+
+def update_recipient_status(report_id, recipient_name, status):
+    """Update the status of a recipient for a report"""
+    conn = create_db_connection()
+    if conn is None:
+        return False
+    
+    try:
+        with conn.cursor() as cur:
+            cur.execute("""
+                UPDATE report_recipients
+                SET status = %s
+                WHERE report_id = %s AND recipient_name = %s;
+            """, (status, report_id, recipient_name))
+            
+            conn.commit()
+            return True
+    except psycopg2.Error as e:
+        st.error(f"Database error when updating recipient status: {e}")
+        conn.rollback()
+        return False
+    finally:
+        conn.close()
+def determine_report_recipients(form_data):
+    """Determine appropriate recipients for the report based on form data"""
+    recipients = []
+    
+    systems = form_data.get("Systems", [])
+    for system in systems:
+        if "OpenAI" in system or "GPT" in system:
+            recipients.append({
```

Comment:
Great! Let's create a separate file with the `class ReportRecipient`. It's fields can be name, type, contact, and an optional submit method?